pkg/model/openstackmodel/servergroup.go

```diff
@@ -96,17 +104,27 @@ func (b *ServerGroupModelBuilder) buildInstances(c *fi.ModelBuilderContext, sg *
 		}
 		c.AddTask(portTask)
 
+		var az *string
+		if ig.Spec.Role == kops.InstanceGroupRoleMaster {
+			// take zone name from ig name, kind of hack. Is there easier way to see az to igs?
+			az = fi.String(strings.Replace(ig.Name, "master-", "", 1))
+		} else {
+			// take index from name, and spread equally to zones
+			az = fi.String(zones[int(i)%len(zones)])
+		}
+
 		instanceTask := &openstacktasks.Instance{
```

Comment:
we have an index to use to mod over all zones.  Would something like this work:

```
		zone := ig.Spec.Zones[ i % len(ig.Spec.Zones)]

		instanceTask := &openstacktasks.Instance{
			Name:        instanceName,
			Region:      fi.String(b.Cluster.Spec.Subnets[0].Region),
			Flavor:      fi.String(ig.Spec.MachineType),
			Image:       fi.String(ig.Spec.Image),
			SSHKey:      fi.String(sshKeyName),
```

Reply:
It does not work. ig.Spec.Zones is empty for all others than GCP

Reply:
see https://github.com/kubernetes/kops/blob/master/cmd/kops/create_cluster.go#L784

Reply:
ofc we could add openstack there as well?